Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -29,7 +65,31 @@ public void AddItem(Item item)
         /// </summary>
         public void RemoveItem(Item item)
         {
+            _items = RemoveFromArray(item);
+        }
+
+        private Item[] RemoveFromArray(Item item)
```

Comment:
Your array operations look neat when looking at them first but I think that they not only add unneccessary complexity but also do more work than you need to do.
I do not see a real benefit here. This is true for all other ones, too, if I am not mistaken.
Also I think - while your basic idea is good - that the results differ from what you were planning to achieve.

Reply:
I have done that because we learned we cannot add remove items from a array. So we need to make a new one. Copy alll the data we want and add or skip the item we want to add or remove.
but because remove is void and  I need to return the new array I need to make another method/function that does that.  
That was my plan and I do not see how it differ from what I was planning to achieve so if you can telll me  where im wrong im happy to change

Reply:
or do you mean I can make one function for adding and one for removing and re-use that for the shop and the player ?